imagenet/main.py

```diff
@@ -290,7 +290,7 @@ def train(train_loader, model, criterion, optimizer, epoch, args):
         # measure data loading time
         data_time.update(time.time() - end)
 
-        if args.gpu is not None:
+        if torch.cuda.is_available():
```

Comment:
i think this should be args.gpu should be existing with regular distributed.
Do you mean it's not working in multiprocessing distributed mode?
https://github.com/pytorch/examples/blob/main/imagenet/main.py#L73-L77

Reply:
The opposite actually. Running with ```--multiprocess-distributed``` sets ```args.gpu``` here: https://github.com/pytorch/examples/blob/main/imagenet/main.py#L119

Running regular distributed (in my case by setting the ```--world-size```, which triggers ```args.distributed``` here: https://github.com/pytorch/examples/blob/main/imagenet/main.py#L102)  leaves ```args.gpu``` unset. 

It looks like this issue has existed for some time, since I found this issue on it with the same suggested fix: https://github.com/pytorch/examples/issues/862

Reply:
If I remember correctly (and I looked at the code more closely now), when you launch distributed yourself (i.e. without relying on multiprocessing distributed), you are responsible for giving `--world-size` and `--gpu` correctly.

For example, the example usage would be:

```
python main.py ... --world-size 2 --rank 0 --gpu 0 imagenet_root &
python main.py ... --world-size 2 --rank 1 --gpu 1 imagenet_root &
```

The patch as you've written it will always effectively run `images = images.cuda(None, non_blocking=True)`, putting the images on GPU0 regardless of which GPU it is supposed to be running on.

Reply:
I think the suggested fix in https://github.com/pytorch/examples/issues/862 is wrong. I'll comment there as well.

Reply:
I see. Thanks for the explanation! Can I create a PR with this example usage in the README so that other users launching regular distributed don't face this issue? 

**Last question:** Would it be helpful for readability to normalize these ```if``` statements? If so, I've updated the PR for this. If not, we can reject this PR entirely.

Reply:
> Can I create a PR with this example usage in the README so that other users launching regular distributed don't face this issue?

Yes